ckanext/data_qld/reporting/controller.py

```diff
@@ -0,0 +1,278 @@
+import logging
+import ckan.plugins.toolkit as toolkit
+
+from ckan.lib.base import BaseController
+from ckanext.data_qld.reporting import constants
+from ckanext.data_qld.reporting.helpers import export_helpers, helpers
+
+log = logging.getLogger(__name__)
+get_action = toolkit.get_action
+request = toolkit.request
+render = toolkit.render
+get_validator = toolkit.get_validator
+ObjectNotFound = toolkit.ObjectNotFound
+Invalid = toolkit.Invalid
+NotAuthorized = toolkit.NotAuthorized
+abort = toolkit.abort
+_ = toolkit._
+DATAREQUEST_OPEN_MAX_DAYS = constants.DATAREQUEST_OPEN_MAX_DAYS
+COMMENT_NO_REPLY_MAX_DAYS = constants.COMMENT_NO_REPLY_MAX_DAYS
+
+
+class ReportingController(BaseController):
+
+    @classmethod
+    def check_user_access(cls):
+        toolkit.check_access(
+            'has_user_permission_for_some_org',
+            helpers.get_context(), {'permission': 'create_dataset'}
+        )
+
+    def index(self):
+        try:
+            self.check_user_access()
+
+            start_date, end_date = helpers.get_report_date_range(request)
+            org_id = request.GET.get('organisation', None)
+
+            organisations = helpers.get_organisation_list()
+
+            if organisations and len(organisations) == 1:
+                org_id = organisations[0]['value']
+
+            extra_vars = {
+                'organisations': organisations,
+                'user_dict': get_action('user_show')({}, {'id': toolkit.c.userobj.id})
+            }
+
+            if org_id:
+                org = get_action('organization_show')({}, {'id': org_id})
```

Comment:
By providing an empty context dictionary, this can only work if organisations are anonymously accessible.

Reply:
As per reply above: You can send an empty context dictionary to an action via the toolkit.get_action method and it will automatically utilise the current context.

btw, when would you encounter a scenario when `organisation_show` is not accessible anonymously on an open data portal?

Reply:
The fact that authorisation checks exist for `organization_show` means we should take into account the possibility of needing them. But if blank contexts are handled, ok.